src/sealteamsixplayer/Mobile.java

```diff
@@ -1,207 +1,275 @@
+    public void go()
+    {
+        super.go();
+
+        try
+        {
+            senseLocations();
+            // Check the blockchain every 5 turns.
+            if (turnCount % 5 == 0)
+                checkBlockchain();
+        }
+        catch (GameActionException e)
+        {
+            e.printStackTrace();
+        }
+
+    }
+
+    /**
+     * Convenience method for checking if we are "at location". That is to say, on top of or next to.
+     */
+    public boolean atLocation(MapLocation target)
+    {
+        return rc.getLocation().equals(target) || rc.getLocation().isAdjacentTo(target);
+    }
+
+    /**
+     * Attempts to move in a given direction.
+     *
+     * @param dir The intended direction of movement
+     * @return
+     */
+    protected boolean tryMove(Direction dir, boolean isDrone) throws GameActionException
+    {
+        if (rc.isReady() && rc.canMove(dir))
+        {
+            if(isDrone)
+            {
+                System.out.println(rc.getType() + " moving " + dir + "--" + rc.getLocation());
+                rc.move(dir);
+                return true;
+            }
+            else
+            {
+                if (!rc.senseFlooding(rc.adjacentLocation(dir)))
+                {
+                    System.out.println(rc.getType() + " moving " + dir + "--" + rc.getLocation());
+                    rc.move(dir);
+                    return true;
+                }
+                return false;
+            }
+        }
+        else
+            return false;
+    }
+
+    /**
+     * Walk in the "general" direction given by <code>dir</code>. Calculates forward as any of:
+     *      forward, left-forward, left, right-forward, right
+     * @returns true if the robot was able to move, false otherwise.
+     */
+    public boolean goTo(Direction dir) throws GameActionException
+    {
+        Direction[] forward = {
+            dir,
+            dir.rotateLeft(),
+            dir.rotateRight(),
+            dir.rotateLeft().rotateLeft(),
+            dir.rotateRight().rotateRight()
+        };
+
+        // Try to move forward.
+        for (Direction d : forward)
+        {
+            if (tryMove(d, isDrone))
+                return true;
+        }
+        System.err.println("Failed to move " + dir + "!");
+        System.err.println("ready:" + rc.isReady() +
+            " can move:" + rc.canMove(dir) +
+            " can sense:" + rc.canSenseLocation(rc.adjacentLocation(dir)) +
+            " robot:" + rc.senseRobotAtLocation(rc.adjacentLocation(dir)) +
+            " flooding:" + rc.senseFlooding(rc.adjacentLocation(dir)));
+        return false;
+    }
+
+    /**
+     * Walk in the "general" direction given by <code>dir</code>. Calculates forward as any of:
+     *      forward, left-forward, left, right-forward, right
+     * @returns true if the robot was able to move, false otherwise.
+     */
+    public boolean goTo(Direction dir, boolean isDrone) throws GameActionException

```

Comment:
The code here looks fine, just wanted to put how I would have done this in case it would be helpful:

1. Modify goTo(Direction dir) to add boolean isDrone, so it is `public boolean goTo(Direction dir, boolean isDrone)` like it is here.
2. Make the following convenience method:
```
public boolean goTo(Direction dir) throws GameActionException
{
    return goTo(dir, false);
}
```

That way you're not copy pasting the majority of the logic and your `isDrone` boolean still gets passed through when needed.